File: proxads/src/main/java/com/proxglobal/proxads/adsv2/admax/MaxNativeAds.java
package com.proxglobal.proxads.adsv2.admax;

import android.app.Activity;
import android.os.Handler;
import android.util.Log;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.applovin.mediation.MaxAd;
import com.applovin.mediation.MaxError;
import com.applovin.mediation.nativeAds.MaxNativeAd;
import com.applovin.mediation.nativeAds.MaxNativeAdListener;
import com.applovin.mediation.nativeAds.MaxNativeAdLoader;
import com.applovin.mediation.nativeAds.MaxNativeAdView;
import com.applovin.mediation.nativeAds.MaxNativeAdViewBinder;
import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdLoader;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.LoadAdError;
import com.google.android.gms.ads.VideoOptions;
import com.google.android.gms.ads.nativead.MediaView;
import com.google.android.gms.ads.nativead.NativeAd;
import com.google.android.gms.ads.nativead.NativeAdOptions;
import com.google.android.gms.ads.nativead.NativeAdView;
import com.proxglobal.proxads.R;
import com.proxglobal.proxads.adsv2.ads.NativeAds;

public class MaxNativeAds extends NativeAds<MaxNativeAdView> {
    private int layoutAdId;

    public MaxNativeAds(Activity activity, FrameLayout container, String adId, int layoutAdId) {
        super(activity, container, adId);
        this.layoutAdId = layoutAdId;
    }

    @Override
    public void specificLoadAdsMethod() {
        MaxNativeAdViewBinder binder = new MaxNativeAdViewBinder.Builder(layoutAdId)
                .setTitleTextViewId(R.id.ad_headline)
                .setBodyTextViewId(R.id.ad_body)
                .setAdvertiserTextViewId(R.id.ad_advertiser)
                .setIconImageViewId(R.id.ad_app_icon)
                .setMediaContentViewGroupId(R.id.ad_media)
                .setOptionsContentViewGroupId(R.id.ad_options_view)
                .setCallToActionButtonId(R.id.ad_call_to_action)
                .build();

        ads = new MaxNativeAdView(binder, mActivity);

        MaxNativeAdLoader nativeAdLoader = new MaxNativeAdLoader(adId, mActivity);
        nativeAdLoader.setNativeAdListener(new MaxNativeAdListener() {
            @Override
            public void onNativeAdLoaded(final MaxNativeAdView nativeAdView, final MaxAd ad) {
                Log.d("ntduc", "onNativeAdLoaded");
                onLoadSuccess();
                if (!mActivity.isDestroyed()) {
                    onShowSuccess();

                    populateNativeAdView(ad.getNativeAd(), nativeAdView);
                    mContainer.removeAllViews();
                    mContainer.addView(nativeAdView);
                }
            }

            @Override
            public void onNativeAdLoadFailed(final String adUnitId, final MaxError error) {
                Log.d("ntduc", "onNativeAdLoadFailed");
                onLoadFailed();
                ads = new MaxNativeAdView(binder, mActivity);
                nativeAdLoader.loadAd(ads);
            }

            @Override
            public void onNativeAdClicked(final MaxAd ad) {
                Log.d("ntduc", "onNativeAdClicked");
                ads = new MaxNativeAdView(binder, mActivity);
                nativeAdLoader.loadAd(ads);
            }
        });

        nativeAdLoader.loadAd(ads);
    }

    @Override
    public void specificShowAdsMethod(Activity activity) {

    }

    public void populateNativeAdView(MaxNativeAd nativeAd, MaxNativeAdView adView) {
        // The headline is guaranteed to be in every UnifiedNativeAd.
        try {
            adView.getTitleTextView().setText(nativeAd.getTitle());
        } catch (Exception e) {
            e.printStackTrace();
        }

        // These assets aren't guaranteed to be in every UnifiedNativeAd, so it's important to
        // check before trying to display them.
        try {
            if (nativeAd.getBody() == null) {
                adView.getBodyTextView().setVisibility(View.INVISIBLE);
            } else {
                adView.getBodyTextView().setVisibility(View.VISIBLE);
                adView.getBodyTextView().setText(nativeAd.getBody());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            if (nativeAd.getAdvertiser() == null) {
                adView.getAdvertiserTextView().setVisibility(View.INVISIBLE);
            } else {
                adView.getAdvertiserTextView().setVisibility(View.VISIBLE);
                adView.getAdvertiserTextView().setText(nativeAd.getAdvertiser());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            if (nativeAd.getIcon() == null) {
                adView.getIconImageView().setVisibility(View.GONE);
            } else {
                adView.getIconImageView().setVisibility(View.VISIBLE);
                adView.getIconImageView().setImageURI(nativeAd.getIcon().getUri());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            if (nativeAd.getMediaView() == null) {
                adView.getMediaContentViewGroup().setVisibility(View.INVISIBLE);
            } else {
                adView.getMediaContentViewGroup().setVisibility(View.VISIBLE);
                adView.getMediaContentViewGroup().removeAllViews();
                adView.getMediaContentViewGroup().addView(nativeAd.getMediaView());

            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            if (nativeAd.getOptionsView() == null) {
                adView.getOptionsContentViewGroup().setVisibility(View.INVISIBLE);
            } else {
                adView.getOptionsContentViewGroup().setVisibility(View.VISIBLE);
                adView.getOptionsContentViewGroup().removeAllViews();
                adView.getOptionsContentViewGroup().addView(nativeAd.getOptionsView());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        try {
            if (nativeAd.getCallToAction() == null) {
                adView.getCallToActionButton().setVisibility(View.INVISIBLE);
            } else {
                adView.getCallToActionButton().setVisibility(View.VISIBLE);
                adView.getCallToActionButton().setText(nativeAd.getCallToAction());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: proxads/src/main/java/com/proxglobal/proxads/adsv2/admax/MaxNativeAds.kt
package com.proxglobal.proxads.adsv2.admax

import com.proxglobal.proxads.adsv2.ads.NativeAds
import android.app.Activity
import android.util.Log
import android.widget.FrameLayout
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import androidx.appcompat.view.ContextThemeWrapper
import com.applovin.mediation.MaxAd
import com.applovin.mediation.MaxError
import com.applovin.mediation.nativeAds.*
import com.proxglobal.proxads.R
import java.lang.Exception

class MaxNativeAds : NativeAds<MaxNativeAdView?> {
    private var layoutAdId: Int
    private var styleBtnAds = 0
    private var isCustomStyle: Boolean

    constructor(
        activity: Activity?,
        container: FrameLayout?,
        adId: String?,
        layoutAdId: Int
    ) : super(activity, container, adId) {
        this.layoutAdId = layoutAdId
        isCustomStyle = false
    }

    constructor(
        activity: Activity?,
        container: FrameLayout?,
        adId: String?,
        layoutAdId: Int,
        styleBtnAds: Int
    ) : super(activity, container, adId) {
        this.layoutAdId = layoutAdId
        this.styleBtnAds = styleBtnAds
        isCustomStyle = true
    }

    override fun specificLoadAdsMethod() {
        val layoutAd = LayoutInflater.from(mActivity).inflate(layoutAdId, mContainer, false)
        var adCallToAction: Button
        try {
            if (isCustomStyle) {
                layoutAd.findViewById<View>(R.id.ad_call_to_action).visibility = View.GONE
                adCallToAction =
                    Button(ContextThemeWrapper(mActivity, styleBtnAds), null, styleBtnAds)
                adCallToAction.layoutParams = LinearLayout.LayoutParams(
                    LinearLayout.LayoutParams.MATCH_PARENT,
                    LinearLayout.LayoutParams.WRAP_CONTENT
                )
                adCallToAction.id = View.generateViewId()
                (layoutAd.findViewById<View>(R.id.linear) as LinearLayout).addView(adCallToAction)
            } else {
                layoutAd.findViewById<View>(R.id.ad_call_to_action).visibility = View.VISIBLE
                adCallToAction = layoutAd.findViewById(R.id.ad_call_to_action)
            }
        } catch (e: Exception) {
            layoutAd.findViewById<View>(R.id.ad_call_to_action).visibility = View.VISIBLE
            adCallToAction = layoutAd.findViewById(R.id.ad_call_to_action)
        }
        val binder = MaxNativeAdViewBinder.Builder(layoutAd)
            .setTitleTextViewId(R.id.ad_headline)
            .setBodyTextViewId(R.id.ad_body)
            .setAdvertiserTextViewId(R.id.ad_advertiser)
            .setIconImageViewId(R.id.ad_app_icon)
            .setMediaContentViewGroupId(R.id.ad_media)
            .setOptionsContentViewGroupId(R.id.ad_options_view)
            .setCallToActionButtonId(adCallToAction.id)
            .build()
        ads = MaxNativeAdView(binder, mActivity)
        val nativeAdLoader = MaxNativeAdLoader(adId, mActivity)
        nativeAdLoader.setRevenueListener { ad ->
            Log.d("ntduc", "NativeMax Revenue: " + ad.revenue)
            Log.d("ntduc", "NativeMax NetworkName: " + ad.networkName)
            Log.d("ntduc", "NativeMax AdUnitId: " + ad.adUnitId)
            Log.d("ntduc", "NativeMax Placement: " + ad.placement)
            Log.d("ntduc", "-------------------------------------------")
        }
        nativeAdLoader.setNativeAdListener(object : MaxNativeAdListener() {
            override fun onNativeAdLoaded(nativeAdView: MaxNativeAdView?, ad: MaxAd) {
                onLoadSuccess()
                if (mActivity.isDestroyed) {
                    nativeAdLoader.destroy()
                } else {
                    onShowSuccess()
                    populateNativeAdView(ad.nativeAd, nativeAdView)
                    mContainer!!.removeAllViews()
                    mContainer!!.addView(nativeAdView)
                }
            }

            override fun onNativeAdLoadFailed(adUnitId: String, error: MaxError) {
                onLoadFailed()
                if (mActivity.isDestroyed) {
                    nativeAdLoader.destroy()
                } else {
                    ads = MaxNativeAdView(binder, mActivity)
                    nativeAdLoader.loadAd(ads)
                }
            }

            override fun onNativeAdClicked(ad: MaxAd) {
                if (mActivity.isDestroyed) {
                    nativeAdLoader.destroy()
                } else {
                    ads = MaxNativeAdView(binder, mActivity)
                    nativeAdLoader.loadAd(ads)
                }
            }
        })
        nativeAdLoader.loadAd(ads)
    }

    override fun specificShowAdsMethod(activity: Activity?) {}
    fun populateNativeAdView(nativeAd: MaxNativeAd?, adView: MaxNativeAdView?) {
        // The headline is guaranteed to be in every UnifiedNativeAd.
        try {
            adView!!.titleTextView.text = nativeAd!!.title
        } catch (e: Exception) {
            e.printStackTrace()
        }

        // These assets aren't guaranteed to be in every UnifiedNativeAd, so it's important to
        // check before trying to display them.
        try {
            if (nativeAd!!.body == null) {
                adView!!.bodyTextView.visibility = View.INVISIBLE
            } else {
                adView!!.bodyTextView.visibility = View.VISIBLE
                adView.bodyTextView.text = nativeAd.body
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        try {
            if (nativeAd!!.advertiser == null) {
                adView!!.advertiserTextView.visibility = View.INVISIBLE
            } else {
                adView!!.advertiserTextView.visibility = View.VISIBLE
                adView.advertiserTextView.text = nativeAd.advertiser
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        try {
            if (nativeAd!!.icon == null) {
                adView!!.iconImageView.visibility = View.GONE
            } else {
                adView!!.iconImageView.visibility = View.VISIBLE
                adView.iconImageView.setImageURI(nativeAd.icon.uri)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        try {
            if (nativeAd!!.mediaView == null) {
                adView!!.mediaContentViewGroup.visibility = View.INVISIBLE
            } else {
                adView!!.mediaContentViewGroup.visibility = View.VISIBLE
                adView.mediaContentViewGroup.removeAllViews()
                adView.mediaContentViewGroup.addView(nativeAd.mediaView)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        try {
            if (nativeAd!!.optionsView == null) {
                adView!!.optionsContentViewGroup.visibility = View.INVISIBLE
            } else {
                adView!!.optionsContentViewGroup.visibility = View.VISIBLE
                adView.optionsContentViewGroup.removeAllViews()
                adView.optionsContentViewGroup.addView(nativeAd.optionsView)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        try {
            if (nativeAd!!.callToAction == null) {
                adView!!.callToActionButton.visibility = View.INVISIBLE
            } else {
                adView!!.callToActionButton.visibility = View.VISIBLE
                adView.callToActionButton.text = nativeAd.callToAction
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}